feature_engine/transformation/ciclycal.py

```diff
@@ -0,0 +1,122 @@
+
+    A list of variables can be passed as an argument. Alternatively, the
+    transformer will automatically select and transform all numerical
+    variables.
+
+    Parameters
+    ----------
+    variables : list, default=None
+        The list of numerical variables that will be transformed. If None, the
+        transformer will automatically find and select all numerical variables.
+
+    Methods
+    -------
+    fit:
+        This transformer does not learn parameters.
+    transform:
+        Apply the CyclicalTransformer transformation.
+    fit_transform:
+        Fit to data, then transform it.
+
+    References
+    ----------
+    ..
+    """
+
+    def __init__(
+            self, variables: Union[None, int, str, List[Union[str, int]]] = None
+    ) -> None:
+        self.variables = _check_input_parameter_variables(variables)
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer does not learn any parameter. But keeps a map of the max values
+        for every column
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y : pandas Series, default=None
+            It is not needed in this transformer. You can pass y or None.
+
+        Raises
+        ------
+        TypeError
+            If the input is not a Pandas DataFrame.
+        ValueError:
+            If some of the columns contains NaNs
+
+        Returns
+        -------
+        self
+        """
+
+        # check input dataframe
+        X = super().fit(X)
+
+        self.input_shape_ = X.shape
+
+        # check for nans
+        self.max_values_ = {}
+        for variable in self.variables:
+            if X[variable].isna().sum() > 0:
+                raise ValueError(f'The transformer CiclycalTransformer '
+                                 f'does not allow to have NaN values, please '
+                                 f'check the column {variable}')
+            self.max_values_[variable] = X[variable].max()
+        return self
+
+    def transform(self, X: pd.DataFrame):
+        """
+        Apply a Ciclycal transformation.
+
+        Parameters
+        ----------
+        X : Pandas DataFrame of shame = [n_samples, n_features]
+            The data to be transformed.
+
+        Raises
+        ------
+        TypeError
+            If the input is not Pandas DataFrame.
+        ValueError:
+            If some of the columns contains NaNs
+        """
+        X = super().transform(X)
+
+        for variable in self.variables:
+            max_value = self.max_values_[variable]
+            X[f'{variable}_sin'] = np.sin(X[variable] * (2. * np.pi / max_value))
+            X[f'{variable}_cos'] = np.cos(X[variable] * (2. * np.pi / max_value))
+            X.drop(columns=variable, inplace=True)
+
```

Comment:
instead of dropping the original features by default, could we add a parameter in init called `drop_original=False`, where the user has the chance to decide if they want to drop or not?

To be honest, I have used months and day of weeks in my predictions, so even if these features add prediction power, I would be hesitant to discard them straightaway.

Reply:
`drop_original` was at the beggining, but I removed it (to be honest I don´t remember by I said no). But Sure! I can bring it to live again.